pkg/apis/istio/v1alpha1/values_types.go

```diff
@@ -0,0 +1,506 @@
+package v1alpha1
+
+// TODO: create remaining enum types.
+
+import (
+	corev1 "k8s.io/api/core/v1"
+)
+
+type Values struct {
+	CertManager     *CertManagerConfig     `json:"certmanager,omitempty"`
+	Galley          *GalleyConfig          `json:"galley,omitempty"`
+	Global          *GlobalConfig          `json:"global,omitempty"`
+	Grafana         map[string]interface{} `json:"grafana,omitempty"`
+	Gateways        *GatewaysConfig        `json:"gateways,omitempty"`
```

Comment:
Shouldn't `gateways` be an slice?  
Or would the intention for those using multiple sets of gateways to create a secondary crd/controller that watches IstioControlPlane.
I'm specifically thinking of the multi-tenant with single control plane use case.

Reply:
agreed, and i think it's a bit more complicated than that.  i'll post a link to work i've done previously to create a go type for values.yaml.

Reply:
This is a struct version of:
https://istio.io/docs/reference/config/installation-options/#gateways-options
The intended purpose is to be able to unmarshal a values.yaml into a struct so it can be validated and passed through to helm charts. 
It's not intended to be a new schema for an installer, that's here: https://github.com/istio/operator/pull/3

Reply:
@ostromart, my point was that the schema for gateways is a map, gateway-name->config, and that the gateways you listed are "standard" gateways typically installed with the control plane, but other gateways can be installed in the same manner, e.g. https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/example-values/values-istio-gateways.yaml#L36

Reply:
Are you saying you updated it to make it generic, if so,`GatewaysConfig` is still a hard-coded list of gateways, which supports some of the known gateways, but is not congruent with the general `values.yaml`.  If you're saying this is something that still needs to be addressed, I'm okay with that.  Let me know.

Reply:
Created https://github.com/istio/istio/issues/14592